utils/check_copies.py

```diff
@@ -226,9 +225,19 @@ def get_indent(code: str) -> str:
     return ""
 
 
-def blackify(code: str) -> str:
+async def run_ruff(code):
+    command = ["ruff", "format", "-", "--config", "pyproject.toml", "--silent"]
+    process = await asyncio.create_subprocess_exec(
+        *command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE
+    )
+    stdout, _ = await process.communicate(input=code.encode())
+    return stdout.decode()
```

Comment:
would love to hear the motivation to make is async 🙏

Reply:
Followed what was done in huggingface hub, nor real reasons apart trying to make it faster. Though it won't be run in parallel. Did not benchmark both

Reply:
Well, as here we don't really use the concurrency (which is the advantage of using `coroutine`), it looks somehow overwhelming to use `async` stuff here.

But my real concern is: here we spawn a new process whenever there is a `# Copied from` statement, which is an expensive op. Previously, `blackify` uses `black.format_str` which is inside the same python process, despite black itself is slow.

I checked the run (check_copies) with this PR and on `main`: 44 sec. on this PR and 32 sec. on `main` .

See [run with this PR](https://app.circleci.com/pipelines/github/huggingface/transformers/78517/workflows/e0cbb6e0-2f1a-4349-a8ea-873d81958a72/jobs/1001612) and [run with main](https://app.circleci.com/pipelines/github/huggingface/transformers/78474/workflows/b3b421a1-aefc-48c5-bcb4-53bff8bc3560/jobs/1000961).

I am afraid, at least for `check_copies.py`, we lose the advantage of using `ruff format` due to the many new processes being created each time.

Reply:
Well I can do not async, no worries. Black is just too complicated, and with ruff you don't need to use doc-builder on top of it. 
Now differences can also come from modified files as well but let's remove black because if you take the `make style` time into account it's already worth it